File: src/#349/solution.js
var intersection = function (nums1, nums2) {
  let map1 = new Map()
  let res = []

  nums1.forEach(num => {
    map1.set(num, 0)
  })

  nums2.forEach(num => {
    if (map1.has(num) && map1.get(num) === 0) {
      res.push(num)
      map1.set(num, 1)
    }
  })

  return res
}

File: src/#349/solution.go
package solution

func intersection(nums1 []int, nums2 []int) []int {
	map1 := map[int]int{}
	res := []int{}
	for _, num := range nums1 {
		map1[num] = 0
	}
	for _, num := range nums2 {
		count, ok := map1[num]
		if ok && count == 0 {
			res = append(res, num)
		}
	}

	return res

}
